Sources/SwiftDocC/Indexing/Navigator/RenderNode+NavigatorIndex.swift

```diff
@@ -0,0 +1,212 @@
+    
+    // Different for each language variant
+    var topicSections: [TaskGroupRenderSection] {
+        wrapped.topicSectionsVariants.value(for: traits)
+    }
+    var defaultImplementationsSections: [TaskGroupRenderSection] {
+        wrapped.defaultImplementationsSectionsVariants.value(for: traits)
+    }
+}
+
+private let typesThatShouldNotUseNavigatorTitle: Set<NavigatorIndex.PageType> = [
+    .framework, .class, .structure, .enumeration, .protocol, .typeAlias, .associatedType, .extension
+]
+
+extension NavigatorIndexableRenderNodeRepresentation {
+    /// Returns a navigator title preferring the fragments inside the metadata, if applicable.
+    func navigatorTitle() -> String? {
+        let tokens: [DeclarationRenderSection.Token]?
+        
+        // FIXME: Use `metadata.navigatorTitle` for all Swift symbols (github.com/apple/swift-docc/issues/176).
+        if identifier.sourceLanguage == .swift || (metadata.navigatorTitle ?? []).isEmpty {
+            let pageType = navigatorPageType()
+            guard !typesThatShouldNotUseNavigatorTitle.contains(pageType) else {
+                return metadata.title
+            }
+            tokens = metadata.fragments
+        } else {
+            tokens = metadata.navigatorTitle
+        }
+        
+        return tokens?.map(\.text).joined() ?? metadata.title
+    }
+    
+    /// Returns the type of page for the render node.
+    func navigatorPageType() -> NavigatorIndex.PageType {
+        // This is a workaround to support plist keys.
+        switch metadata.roleHeading?.lowercased() {
+            case "property list key":           return .propertyListKey
+            case "property list key reference": return .propertyListKeyReference
+            default: break
+        }
+        
+        switch kind {
+            case .article:  return metadata.role.map { .init(role: $0) }
+                                ?? .article
+            case .tutorial: return .tutorial
+            case .section:  return .section
+            case .overview: return .overview
+            case .symbol:   return metadata.symbolKind.map { .init(symbolKind: $0) }
+                                ?? metadata.role.map { .init(role: $0) }
+                                ?? .symbol
+        }
+    }
+}
+
+extension NavigatorIndexableRenderNodeRepresentation {
+    func navigatorChildren(for traits: [RenderNode.Variant.Trait]?) -> [RenderRelationshipsGroup] {
+        switch kind {
+        case .overview:
+            var groups = [RenderRelationshipsGroup]()
+            for case let section as VolumeRenderSection in sections {
+                groups.append(contentsOf: section.chapters.map { chapter in
+                    RenderRelationshipsGroup(
+                        name: chapter.name,
+                        abstract: nil,
+                        references: chapter.tutorials.compactMap { self.references[$0.identifier] as? TopicRenderReference }
```

Comment:
Ok, makes sense. I was initially thinking it might be useful to have a warning or something here so children weren't silently dropped from the navigator, but it sounds like that's probably not a situation we have to worry about right now.